src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultKeyValueAdapter.cs

```diff
@@ -102,5 +92,50 @@ public bool NeedsRefresh()
         {
             return _secretProvider.ShouldRefreshKeyVaultSecrets();
         }
+
+        private string ParseSecretReferenceUri(ConfigurationSetting setting)
+        {
+            string secretRefUri = null;
+
+            try
+            {
+                var reader = new Utf8JsonReader(System.Text.Encoding.UTF8.GetBytes(setting.Value));
+
+                if (reader.Read() && reader.TokenType != JsonTokenType.StartObject)
+                {
+                    throw CreateKeyVaultReferenceException("Invalid Key Vault reference.", setting, null, null);
+                }
+
+                while (reader.Read() && reader.TokenType != JsonTokenType.EndObject)
+                {
+                    if (reader.TokenType != JsonTokenType.PropertyName)
+                    {
+                        continue;
+                    }
+
+                    if (reader.GetString() == KeyVaultConstants.SecretReferenceUriJsonPropertyName)
+                    {
+                        if (reader.Read() && reader.TokenType == JsonTokenType.String)
+                        {
+                            secretRefUri = reader.GetString();
+                        }
+                        else
+                        {
+                            throw CreateKeyVaultReferenceException("Invalid Key Vault reference.", setting, null, null);
+                        }
+                    }
+                    else
+                    {
+                        reader.Skip();
+                    }
+                }
+            }
+            catch (JsonException e)
```

Comment:
So probably just catch and throw `FormatException` like we do with `JsonException`?

Reply:
That makes sense for feature flags. For KeyVault refs, we dont throw any FormatException, right? So just throw KeyVaultReferenceException here.

Reply:
Oops didn't see the file - not formatexception, thanks

Reply:
I couldn't see a way to add a test for this, not sure if there's a way to pass a string value that will generate invalid utf8 bytes after encoding.

Reply:
Yea we do. Should I remove the catch for InvalidOperationException then? Seems like it shouldn't happen since we have checks for string type before each GetString call and we encode the bytes ourselves.

Reply:
That's a great point! So we dont need to handle InvalidOperationException.